### PR TITLE
Reduce compilation of Base methods (NFC)

### PR DESCRIPTION
**Pull request by timholy**
This is motivated by https://github.com/JuliaLang/julia/pull/43990#issuecomment-1025692379,
but doesn't seem likely to hurt anything even if that never merges.

### CODE REVIEW

File: src/toplevel_generators.jl [hunk 1]
```diff
@@ -133,10 +133,14 @@ function generate_wrapper_load(src_name, pkg_uuid, __source__)
             # Helper function to parse triplets for us
             $(platform_parse_compat())
             function make_wrapper_dict(dir, x)
-                return Dict(
-                    parse_wrapper_platform(basename(f)[1:end-3]) =>
-                        joinpath(dir, "wrappers", f) for f in x
-                )
+                # (1) make the Dict type inferrable
```
Comment (giordano) [inline src/toplevel_generators.jl:136]: For this point, using `Dict{Platform,String}( <generator> )` would be sufficient, right?

**Review by giordano — APPROVED** [review]
With Julia v1.7.1, on current `master` (same as `v1.4.0`):
```julia
% julia --project=/tmp --startup-file=no -q
julia> @time using GTK3_jll
  0.106958 seconds (367.35 k allocations: 17.844 MiB, 23.89% compilation time)
```

With this PR:
```julia
% julia --project=/tmp --startup-file=no -q
julia> @time using GTK3_jll
  0.102986 seconds (363.60 k allocations: 17.562 MiB, 25.75% compilation time)
```

It seems to be a net positive improvement in terms of memory allocations (even if very little) :+1:

File: src/toplevel_generators.jl (hunk 1 above)
Comment (timholy) [reply src/toplevel_generators.jl:136]: Yes, specifying the exact `Dict` type would fix that.

**Comment by timholy** [comment]
Any objections to me making a release?

**Comment by giordano** [comment]
No, it's fine :+1:

**Pull request merged**